tests/scripts/all.sh

```diff
@@ -555,6 +552,26 @@ pre_check_tools () {
     "$@" scripts/output_env.sh
 }
 
+need_submodule () {
+    case $RUN_COMPONENTS in
+        *test_submodule*) return 0;;
```

Comment:
This is not correct, as `component_test_use_psa_crypto_full_cmake_asan` relies on the submodule as well.

Also, in the near future (#2403 and hopefully soon after that, de-duplicating files) all tests will require the submodule, so do we really want to invest time in making `all.sh` working without it now?